### PR TITLE
Removes head_tag_content override to pick up Google Scholar tags from Sufia. Fixes #620.

### LINKED ISSUE #620: As a data curator and product owner I would like to automatically have proxy for all users so that I can assist in uploading and curating their data without too much hassle.
Not  sure this is doable, but I think it would make things more seamless especially for large file uploads.

### PR DESCRIPTION
**Pull request by mjgiarlo**
This work required first getting ScholarSphere working on top of Sufia 4.1.0, which included proxy- and transfer-related code ported from ScholarSphere, so this meant a lot of deletion and cleanup. Then I needed to tweak Sufia to allow downstream head content injection, which is sitting in a branch awaiting a PR merge here: https://github.com/projecthydra/sufia/pull/714 The last step was creating a new override in ScholarSphere to inject the favicon_link_tag that isn't in Sufia. A new spec test is included to make sure the favicon exists.

:bangbang: If you want to wait for the Sufia PR to be closed first so our Gemfile points at the master branch rather than a short-lived feature branch, that's cool -- I can tweak the Gemfile and Gemfile.lock once that happens, squash, and force-push the changes here.

### CODE REVIEW

**Comment by awead** [comment]
@mjgiarlo I merged projecthydra/sufia#714 so I'd go ahead and update this PR.

**Comment by mjgiarlo** [comment]
Will do!

**Comment by mjgiarlo** [comment]
@awead Updated!

**Comment by awead** [comment]
So much deleted code! :heart_eyes:

File: Gemfile [hunk 1]
```diff
@@ -5,7 +5,8 @@ gem 'rails', '4.1.7'
 gem 'mysql2', '0.3.16'
 
 # Hydra community components
-gem 'sufia', '~> 4.0.1'
+#gem 'sufia', '~> 4.2'
```
Comment (jcoyne) [inline Gemfile]: remove this line

Comment (mjgiarlo) [inline Gemfile]: Done.

**Comment by mjgiarlo** [comment]
All requested changes made.

**Comment by mjgiarlo** [comment]
Restarted subfeature specs because they were wonky on Travis. Given how little changed between when specs passed and now, I'm thinking the errors are false positives due to something weird in the CI environment.  Restarting the failed test.

**Pull request merged**